orderer/consensus/etcdraft/storage_test.go

```diff
@@ -395,3 +395,82 @@ func TestApplyOutOfDateSnapshot(t *testing.T) {
 		assertFileCount(t, 12, 1)
 	})
 }
+
+func TestAbortWhenWritingSnapshot(t *testing.T) {
```

Comment:
I add the unit test to reproduce the problem.

Reply:
Thanks but I meant a unit test that uses an instance of a Fabric Raft chain, in `orderer/consensus/etcdraft/chain.go`,  not a unit test that uses pure etcd.io/raft packages. 

We need to be sure that a Fabric Raft chain instance can encounter the bespoken problem, so that:
1. We know we really have a problem, because it might be that Fabric sidesteps this problem via some mechanism and etcd has this problem.
2. If the problem occurs in a later point due to a code change, a unit test will notify us.

Reply:
The problem can be reproduced with the following steps:

1. Orderer A has not participated in consensus for a while.

2. Other orderers generate new blocks in consensus.

3. Other orderers generate a new snapshot.

4. Orderer A back to normal and receives the new snapshot from other orderers.

5. Orderer A persists the new snapshot but crashes before calling `rs.saveSnap(snapshot)`.

6. Orderer A restarts.